### PR TITLE
CLID-515: claude: add slash command to generate test cases and testing approach based on PR.

### PR DESCRIPTION
**Pull request by nidangavali**
# Description

Add a `/test-pr `  claude command to generate test cases and testing approaches based on the respective PR.

Github / Jira issue: [CLID-515](https://issues.redhat.com/browse/CLID-515)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code Improvements (Refactoring, Performance, CI upgrades, etc)
- [x] Internal repo assets (diagrams / docs on github repo)
- [ ] This change requires a documentation update on openshift docs

# How Has This Been Tested?

Run `/test-pr pr-url `when using claude.

## Expected Outcome

  1. PR Change Analysis — Fetches the PR diff using git and summarizes:
    - Files modified
    - Scope of changes (new features, bug fixes, refactoring)
    - Code patterns (functions added/modified)
    - Whether test files are already included
  2. Interactive Test Type Selection — Asks you which test types are needed (multi-select):
    - Unit, Integration, E2E, Regression, Manual, Ad-hoc, or Exploratory
  3. Targeted Testing Guidance — For each selected test type, provides:
    - Unit Tests: Lists testable functions with file paths/line numbers, prioritization, edge cases, mock
  requirements, and a Go table-driven test template
    - Integration Tests: Identifies integration points, component interactions, and run commands
    - E2E Tests: Identifies affected user workflows (M2D, D2M, M2M) with prerequisites and commands
    - Regression Tests: Lists affected code paths and existing test suites to run
    - Manual Tests: Generates step-by-step test case checklists
    - Ad-hoc/Exploratory: Provides exploration guidelines and boundary conditions
  4. Detailed Steps per Function/Scenario — For each item you choose to test:
    - Test file location
    - Starter test template (Go code)
    - Specific go test run commands (with coverage, race detection)
    - Validation criteria
  5. Iterative Loop — After each function/scenario, asks what to do next: test another function, move to the
  next test type, or finish.

  Key point: It does not write actual test code unless you explicitly ask — it provides templates, guidance, and
   commands for you to execute.

### CODE REVIEW

**Review by adolfo-ab — COMMENTED** [review]
I think we should avoid talking about "QE testing", since we are trying to bridge the gap between dev and QE, and instead talk only about testing from a unified point of view

**Comment by nidangavali** [comment]
> I think we should avoid talking about "QE testing", since we are trying to bridge the gap between dev and QE, and instead talk only about testing from a unified point of view

Agreed but the reason I added QE-focus was because of the areas the slash command will mostly cover. I tried to avoid unit tests majorly for 2 reasons:
1. I feel the unit test should be more precisely focused on the code because of which there should be a separate slash command for unit test.
2.The output given by the current slash command is very detailed and long. Adding output for unit test which will also involve multiple code snippets of tests in case there are multiple changes in the PR.
This slash command mainly focuses on giving the testing approach and not the code for automation or any code changes.

File: .claude/commands/README.md [hunk 1]
```diff
@@ -136,6 +136,303 @@ The command will:
...
 
+### `/pr-tests`
+
+Intelligent testing recommendations for your Pull Requests, powered by Claude Code.
+
+An advanced Claude Code skill that analyzes GitHub Pull Requests and generates comprehensive, context-aware testing strategies tailored to your code changes.
+
+### Why Use This?
+
+- **QE-Focused Analysis** - Tailored for Quality Engineering.
```
Comment (aguidirh) [inline .claude/commands/README.md]: Instead of QE-Focused which is not very specific, I would suggest to use something like the type of test this is focused since tests should be everyone responsability.

File: .claude/commands/README.md [hunk 2]
```diff
@@ -136,6 +136,303 @@ The command will:
...
+| **Test Environment Guidance** | Setup requirements, test data needs, and configuration |
+
+## Installation
+
+### Prerequisites
+
+
+1. Authenticate GitHub CLI:
+```bash
+gh auth login
```
Comment (aguidirh) [inline .claude/commands/README.md]: Since our repositories are public, gh is not necessary to check the PRs, only git is enough to pull and do the diff on the PR. In this way we avoid adding an uneeded dependency on the slash command. Git usually is already installed in everyone machine, gh not necessarily.

File: .claude/commands/README.md [hunk 3]
```diff
@@ -136,6 +136,303 @@ The command will:
...
+### Input
+
+```bash
+/pr-tests https://github.com/openshift/oc-mirror/pull/1289
+```
+
+### Output
+
+```markdown
+● 🧪 QE PR Testing Strategy Analysis
```
Comment (aguidirh) [inline .claude/commands/README.md]: Instead of saying QE strategy, it would be nice if we could have strategies by test categories. 

For example, integration test strategy analysis, regression test strategy analysis, etc.

In this way we make clear for the entire team what are the activities that QE is doing today so it is possible to get help from everyone because the steps are more transparent and easy to understand.

File: .claude/commands/README.md [hunk 4]
```diff
@@ -136,6 +136,303 @@ The command will:
...
+  Configuration Needed for Testing:
+  - Multiple working directory configurations
+  - Different log level settings (debug, info, warning, error)
+  - Various oc-mirror command configurations
+  - CI/CD pipeline integration configurations
+
+  Files Not Requiring Tests
+
+  N/A - Only one source file was modified and it requires comprehensive QE testing due to the behavioral change in log management.
+
```
Comment (aguidirh) [inline .claude/commands/README.md]: The change analysis from line 236-419 could be simplified by test categories. Like what needs to be tested on the integration category? What needs to be tested from the regression category? In this way it is easier to be reproducible by everyone.

**Review by aguidirh — CHANGES_REQUESTED** [review]
Hi @nidangavali,

Thanks for submitting this PR, I added some suggestions to your PR. Feel free to ask question about them.

**Review by aguidirh — CHANGES_REQUESTED** [review]
@nidangavali I think you need to rebase this PR and sign the commit.

Also we don't have `.claude` directory anymore, we decided to have something more generic like `.agents` so we can change the AI enginee later for another one if needed. 

/assign @adolfo-ab could you please have a look if this slash command is aligned with our current test strategy?

**Comment by nidangavali** [comment]
@aguidirh I will rebase the PR incase the slash command is required. @adolfo-ab Please let me know if this is relevant or not as per current standards otherwise I will close this PR. Thank You!
